changelog/4321.breaking.rst

```diff
@@ -0,0 +1 @@
+Removed :meth:`~sunpy.net.dataretriever.GenericClient._makeargs`, :meth:`~sunpy.net.dataretriever.GenericClient._get_url_for_timerange`, :meth:`~sunpy.net.dataretriever.GenericClient._makeimap`, :meth:`~sunpy.net.dataretriever.GenericClient_get_time_for_url.`, :meth:`~sunpy.net.dataretriever.GenericClient._link`.
```

Comment:
We don't need a changelog entry for private (`_`) methods.

Reply:
Ok, removed.